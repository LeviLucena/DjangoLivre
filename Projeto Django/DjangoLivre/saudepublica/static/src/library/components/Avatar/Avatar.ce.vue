<template>
  <span
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    class="br-avatar"
    :class="[
      {
        small: density === 'small',
        medium: density === 'medium',
        large: density === 'large',
      },
    ]"
  >
    <span v-if="isIcon || image" class="image">
      <IconBase v-if="isIcon" icon-name="user"></IconBase>
      <img v-else :src="image" alt="Avatar" loading="lazy" />
    </span>
    <span v-else class="image bg-support-01 text-secondary-01">
      {{ avatarLetter }}
    </span>
  </span>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"
import IconBase from "../IconBase/IconBase.ce.vue"

const Avatar = {
  name: "brAvatar",
  components: {
    IconBase,
  },
  mixins: [TooltipMixin],
  props: {
    /**
     * Nome do usuário que será apresentado no avatar pela letra
     */
    name: {
      type: String,
      default: "John Doe",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o avatar apresentado será do tipo fotográfico
     */
    image: {
      type: String,
      validator: function (val) {
        return val.match(/^https?:\/\/[\w\d/.]*/g)
      },
      default: null,
    },
    /**
     * Se informado define que o avatar apresentado será do tipo ícone
     */
    iconic: {
      type: Boolean,
      default: false,
    },
    /**
     * Niveis de densidade para o elemento avatar, opções validas são "small",
     * "medium" e "large" e "smallint", "biginyt" e "default"
     */
    density: {
      type: String,
      default: "small",
      validator: function (value) {
        return ["small", "medium", "large"].includes(value)
      },
    },
  },
  computed: {
    isIcon() {
      return this.iconic && !this.image
    },
    avatarLetter() {
      return this.name[0].toUpperCase()
    },
  },
}

export default Avatar
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/avatar/_avatar";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
