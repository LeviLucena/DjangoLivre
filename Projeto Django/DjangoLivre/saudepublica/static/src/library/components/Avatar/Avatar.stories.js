import { kebabiseArgs } from '../../util/Utils.js'
import BrAvatar from './Avatar.ce.vue'

export default {
  title: 'Componentes/Avatar',
  component: BrAvatar,
  parameters: {
    actions: {
      handles: ['click'],
    },
  },
  argTypes: {
    name: {
      control: 'text',
    },
    iconic: {
      control: 'boolean',
    },
    density: {
      options: ['small', 'medium', 'large'],
      control: { type: 'inline-radio' },
    },
    tooltipText: {
      control: 'text',
    },
    tooltipPlace: {
      options: ['top', 'right', 'bottom', 'left'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipType: {
      options: ['info', 'success', 'warning', 'error'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipTimer: {
      control: 'number',
      if: { arg: 'tooltipText', truthy: true },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-avatar v-bind="args"></br-avatar>`,
})

export const Iconico = Template.bind({})
Iconico.args = {
  name: 'Jhon Doe',
  iconic: true,
}

export const Fotografico = Template.bind({})
Fotografico.args = {
  name: 'Jhon Doe',
  iconic: false,
  image: 'https://picsum.photos/id/823/400',
}

export const Letra = Template.bind({})
Letra.args = {
  name: 'Jhon Doe',
  iconic: false,
}

export const ComTooltip = Template.bind({})
ComTooltip.args = {
  name: 'Jhon Doe',
  iconic: false,
  tooltipText: 'Lorem ipsum',
  tooltipPlace: 'right',
}
