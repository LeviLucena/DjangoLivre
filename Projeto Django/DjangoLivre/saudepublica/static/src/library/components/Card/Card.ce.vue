<template>
  <div
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    class="br-card"
    :class="[{ disabled: disabled, hover: hover, 'h-fixed': hFixed }]"
    :disabled="disabled"
    :aria-disabled="disabled"
  >
    <div class="card-header">
      <slot name="header"></slot>
    </div>
    <div class="card-content" tabindex="0">
      <slot name="content"></slot>
    </div>
    <div class="card-footer">
      <slot name="footer"></slot>
    </div>
  </div>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"

const BrCard = {
  name: "brCard",
  mixins: [TooltipMixin],
  props: {
    /**
     * Aplica estilo desabilitado ao componente
     */
    disabled: {
      type: Boolean,
      default: null,
    },
    /**
     * Habilita o estado hover ao card
     */
    hover: {
      type: Boolean,
      default: null,
    },
    /**
     * Habilita altura fixa e ativa barra de rolagem se necess√°rio
     */
    hFixed: {
      type: Boolean,
      default: false,
    },
  },
}

export default BrCard
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/utilities/_states";
@import "~@govbr-ds/core/src/components/card/_card";
</style>
