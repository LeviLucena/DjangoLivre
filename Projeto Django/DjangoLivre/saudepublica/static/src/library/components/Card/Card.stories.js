import { action } from '@storybook/addon-actions'
import { kebabiseArgs } from '../../util/Utils.js'
import BrCard from './Card.ce.vue'
import CardHeader from './CardHeader.ce.vue'

let icon = 'chevron-down'

const actionsClick = {
  onClick: action('click'),
}

const slotContent = `
    <br-card-content slot="content">
      <img src="https://picsum.photos/id/0/500" alt="Imagem de exemplo"/>
    </br-card-content>
`

const slotContent2 = `
    <br-card-content slot="content">
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit.
        Tempore perferendis nam porro atque ex at, numquam non optio ab
        eveniet error vel ad exercitationem, earum et fugiat recusandae
        harum? Assumenda.
      </p>
    </br-card-content>
`

const slotHeader = `
    <br-card-header slot="header">
      <div class="d-flex">
        <br-avatar image="https://picsum.photos/id/823/400"></br-avatar>
        <div class="ml-3">
          <div class="text-weight-semi-bold text-up-02">Maria Amorim</div>
          <div>UX Designer</div>
        </div>
        <div class="ml-auto">
          <br-button circle icon="ellipsis-v" @click="onClick"></br-button>
        </div>
      </div>
    </br-card-header>
`

const slotHeaderDisabled = `
    <br-card-header slot="header">
      <div class="d-flex">
        <br-avatar image="https://picsum.photos/id/823/400" @click="onClick"></br-avatar>
        <div class="ml-3">
          <div class="text-weight-semi-bold text-up-02">Maria Amorim</div>
          <div>UX Designer</div>
        </div>
        <div class="ml-auto">
          <br-button circle icon="ellipsis-v" disabled></br-button>
        </div>
      </div>
    </br-card-header>
`

const slotFooter = `
    <br-card-footer slot="footer">
      <div class="d-flex">
        <div>
          <br-button label="Button" @click="onClick"></br-button>
        </div>
        <div class="ml-auto">
          <br-button circle icon="share-alt" @click="onClick"></br-button>
          <br-button circle icon="heart" @click="onClick"></br-button>
        </div>
      </div>
    </br-card-footer>
`

const slotFooterDisabled = `
    <br-card-footer slot="footer">
      <div class="d-flex">
        <div>
          <br-button label="Button" disabled></br-button>
        </div>
        <div class="ml-auto">
          <br-button circle icon="share-alt" disabled></br-button>
          <br-button circle icon="heart" disabled></br-button>
        </div>
      </div>
    </br-card-footer>
`

const slotHeaderComTooltip = `
    <br-card-header slot="header">
      <div class="d-flex">
        <br-avatar image="https://picsum.photos/id/823/400" tooltip-text="Lorem ipsum dolor sit amet." @click="onClick"></br-avatar>
        <div class="ml-3">
          <div class="text-weight-semi-bold text-up-02">Maria Amorim</div>
          <div>UX Designer</div>
        </div>
        <div class="ml-auto">
          <br-button circle icon="ellipsis-v" tooltip-text="Lorem ipsum dolor sit amet." @click="onClick"></br-button>
        </div>
      </div>
    </br-card-header>
`

const slotFooterComTooltip = `
<br-card-footer slot="footer">
  <div class="d-flex">
    <div>
      <br-button label="Button" @click="onClick"></br-button>
    </div>
    <div class="ml-auto">
      <br-button circle icon="share-alt" tooltip-text="Lorem ipsum dolor sit amet." @click="onClick"></br-button>
      <br-button circle icon="heart" tooltip-text="Lorem ipsum dolor sit amet." @click="onClick"></br-button>
    </div>
  </div>
</br-card-footer>
`

const slotFooterComCollapse = `
  <br-card-footer slot="footer">
    <div class="text-right">
      <br-button circle icon="${icon}" type="tertiary" id="item-01" @click="this.toggle()"></br-button>
    </div>
    <br-collapse id="br-item-collapse" @toggle-collapse="handleCollapse">
      <br-list>
        <br-item>
          <br-list>
            <br-item>
              <div class="row">
                <div class="col-auto"><i class="fas fa-heartbeat" aria-hidden="true"></i></div>
                <div class="col">
                  <p class="m-0">Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
                </div>
              </div>
            </br-item>
            <br-divider></br-divider>
            <br-item>
              <div class="row">
                <div class="col-auto"><i class="fas fa-heartbeat" aria-hidden="true"></i>
                </div>
                <div class="col">
                  <p class="m-0">Nesciunt, laudantium ea officiis hic tempora velit aperiam nemo accusamus nisi, eligendi ducimus! Incidunt ullam minima ratione amet sequi.</p>
                </div>
              </div>
            </br-item>
            <br-divider></br-divider>
            <br-item>
              <div class="row">
                <div class="col-auto"><i class="fas fa-heartbeat" aria-hidden="true"></i>
                </div>
                <div class="col">
                  <p class="m-0">Voluptates, iste recusandae.</p>
                </div>
              </div>
            </br-item>
          </br-list>
        </br-item>
      </br-list>
    </br-collapse>
  </br-card-footer>
`

export default {
  title: 'Componentes/Card',
  component: BrCard,
  subcomponents: {
    'br-card-header': CardHeader,
  },
  argTypes: {
    header: {
      control: 'text',
      description: '** header slot - exclusivo para títulos, subtítulos, ícones, avatares e tags.',
      table: {
        type: {
          summary: 'Element <br-card-header slot="header">',
        },
        defaultValue: {
          summary: '',
        },
      },
    },
    content: {
      control: 'text',
      description:
        '** content slot - qualquer componente é aceitável, exceto componentes relacionados à navegação, como: carrossel, paginação, abas e menu.',
      table: {
        type: {
          summary: 'Element <br-card-content slot="content">',
        },
        defaultValue: {
          summary: '',
        },
      },
    },
    footer: {
      control: 'text',
      description: '** footer slot - exclusivo para botões e links.',
      table: {
        type: {
          summary: 'Element <br-card-footer slot="footer">',
        },
        defaultValue: {
          summary: '',
        },
      },
    },
    disabled: {
      defaultValue: false,
    },
    hover: {
      defaultValue: false,
    },
    hFixed: {
      defaultValue: false,
    },
  },
}

const TemplateContent = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `
  <div class="col-sm-6 col-md-4 col-lg-3">
    <br-card v-bind="args">${args.content}</br-card>
  </div>
  `,
})

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `
  <div class="col-sm-6 col-md-4 col-lg-3">
    <br-card v-bind="args">${args.header}${args.content}${args.footer}</br-card>
  </div>
  `,
  methods: actionsClick,
})

const TemplateHeader = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<div class="col-sm-6 col-md-4 col-lg-3">
  <br-card v-bind="args">${args.header}  </br-card>
</div>
  `,
  methods: actionsClick,
})

const TemplateFooter = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<div class="col-sm-6 col-md-4 col-lg-3">
  <br-card v-bind="args">${args.footer}  </br-card>
</div>
  `,
  methods: actionsClick,
})

const TemplateFooterComCollapse = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<div class="col-sm-6 col-md-4 col-lg-3">
  <br-card v-bind="args">${args.header}${args.content}${args.footer}  </br-card>
</div>
  `,
  methods: {
    toggle() {
      // this.$refs['collapse'].toggle()
      console.log('toggle()')

      const item = document?.getElementById('item-01')
      const itemCollapse = document?.getElementById('br-item-collapse')

      // TODO: Usar actions nessa parte do projeto para acionar via actions esse metódo
      if (itemCollapse?.hasAttribute('collapsed') === false) {
        itemCollapse.setAttribute('collapsed', true)
        if (item?.icon) item.icon = 'chevron-up'
      } else {
        itemCollapse?.removeAttribute('collapsed')
        if (item?.icon) item.icon = 'chevron-down'
      }
    },
    handleCollapse: action('toggle-collapse'),
    onClick: action('click'),
  },
})

export const Simples = TemplateContent.bind({})
Simples.parameters = { controls: { exclude: ['header'] } }
Simples.args = {
  content: slotContent,
}

export const HeaderSlot = TemplateHeader.bind({})
HeaderSlot.args = {
  header: slotHeader,
}

export const ContentSlot = TemplateContent.bind({})
ContentSlot.args = {
  content: slotContent2,
}
//ContentSlot.parameters = { controls: { exclude: ['slotTemplate'] } }

export const FooterSlot = TemplateFooter.bind({})
FooterSlot.args = {
  footer: slotFooter,
}
//FooterSlot.parameters = { controls: { exclude: ['slotTemplate'] } }

export const Completo = Template.bind({})
Completo.args = {
  header: slotHeader,
  content: slotContent2,
  footer: slotFooter,
}
//Completo.parameters = { controls: { exclude: ['slotTemplate'] } }

export const Disabled = Template.bind({})
Disabled.args = {
  disabled: true,
  header: slotHeaderDisabled,
  content: slotContent2,
  footer: slotFooterDisabled,
}
//Disabled.parameters = { controls: { exclude: ['slotTemplate'] } }

export const Hover = Template.bind({})
Hover.args = {
  hover: true,
  header: slotHeader,
  content: slotContent2,
  footer: slotFooter,
}

export const ComTooltip = Template.bind({})
ComTooltip.args = {
  header: slotHeaderComTooltip,
  content: slotContent2,
  footer: slotFooterComTooltip,
}

export const ComCollapse = TemplateFooterComCollapse.bind({})
ComCollapse.args = {
  header: slotHeader,
  content: slotContent2,
  footer: slotFooterComCollapse,
  // onclick: itemClickCollapsed,
}
