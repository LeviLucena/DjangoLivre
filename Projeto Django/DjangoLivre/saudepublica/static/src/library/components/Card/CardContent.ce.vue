<template>
  <slot></slot>
</template>

<script>
const CardContent = {
  name: "brCardContent",
  props: {},
}

export default CardContent
</script>
