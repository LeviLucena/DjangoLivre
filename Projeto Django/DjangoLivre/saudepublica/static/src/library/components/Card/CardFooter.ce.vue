<template>
  <slot></slot>
</template>

<script>
const CardFooter = {
  name: "brCardFooter",
  props: {},
}

export default CardFooter
</script>
