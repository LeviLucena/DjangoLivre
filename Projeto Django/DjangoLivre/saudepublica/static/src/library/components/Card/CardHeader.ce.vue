<template>
  <slot :disabled="disabled"></slot>
</template>

<script>
const CardHeader = {
  name: "brCardHeader",
  props: {
    /**
     * Aplica estilo desabilitado ao componente
     */
    disabled: {
      type: Boolean,
      default: null,
    },
  },
}

export default CardHeader
</script>
