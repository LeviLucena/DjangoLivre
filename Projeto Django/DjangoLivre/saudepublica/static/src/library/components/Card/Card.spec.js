import { shallowMount } from '@vue/test-utils'

import Avatar from '../Avatar/Avatar.ce.vue'
import Card from '../Card/Card.ce.vue'
import CardContent from '../Card/CardContent.ce.vue'
import CardFooter from '../Card/CardFooter.ce.vue'
import CardHeader from '../Card/CardHeader.ce.vue'
import BrCollapse from '../Collapse/Collapse.ce.vue'

describe('Card', () => {
  test('Testa se a prop disable está sendo renderizada.', () => {
    const wrapper = shallowMount(Card, {
      propsData: {
        disable: true,
      },
    })

    expect(wrapper.find('.br-card').attributes('disable')).toBe('true')
  })

  test('Testa a renderização do card com Collapse.', () => {
    const wrapper = shallowMount(Card, {
      slots: {
        header: `
        <br-card-footer slot="footer">
        <br-collapse show-icon :collapsed="collapse">
          <div class="br-list mt-3">
            <div class="br-item">
              <div class="row">
                <div class="col-auto"><i class="fas fa-heartbeat" aria-hidden="true"></i></div>
                <div class="col">
                  <p class="m-0">Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
                </div>
              </div>
            </div>
            <span class="br-divider"></span>
            <div class="br-item">
              <div class="row">
                <div class="col-auto"><i class="fas fa-heartbeat" aria-hidden="true"></i></div>
                <div class="col">
                  <p class="m-0">
                    Nesciunt, laudantium ea officiis hic tempora velit aperiam nemo accusamus nisi, eligendi ducimus!
                    Incidunt ullam minima ratione amet sequi.
                  </p>
                </div>
              </div>
            </div>
            <span class="br-divider"></span>
            <div class="br-item">
              <div class="row">
                <div class="col-auto"><i class="fas fa-heartbeat" aria-hidden="true"></i></div>
                <div class="col">
                  <p class="m-0">Voluptates, iste recusandae.</p>
                </div>
              </div>
            </div>
          </div>
        </br-collapse>
      </br-card-footer>
        `,
      },
      global: {
        stubs: {
          'br-card-footer': CardFooter,
          'br-collapse': BrCollapse,
        },
      },
    })

    expect(wrapper.findComponent(CardFooter))
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa a renderização do slot header.', () => {
    const wrapper = shallowMount(Card, {
      slots: {
        header: `
          <div slot="header">
            <br-card-header >
              <div class="d-flex">
                <br-avatar imgSrc="https://picsum.photos/id/823/400" />
                <div class="ml-3">
                  <p class="h5 text-primary-default mb-0">Maria Amorim</p>
                  <span>UX Designer</span>
                </div>
                <div class="ml-auto">
                  <button type="circle">
                    <span slot="button-icon">
                      <i class="fas fa-ellipsis-v" aria-hidden="true"></i>
                    </span>
                  </button>
                </div>
              </div>
            </br-card-header>
          </div>
        `,
      },
      global: {
        stubs: {
          'br-card-header': CardHeader,
          'br-avatar': Avatar,
        },
      },
    })

    expect(wrapper.findComponent(CardHeader))
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa a renderização do slot content.', () => {
    const wrapper = shallowMount(Card, {
      slots: {
        content: `
        <div slot="content">
          <br-card-content>
            <p>
              Lorem ipsum dolor sit, amet consectetur adipisicing elit.
              Tempore perferendis nam porro atque ex at, numquam non
              optio ab eveniet error vel ad exercitationem, earum et
              fugiat recusandae harum? Assumenda.
            </p>
          </br-card-content>
        </div>
        `,
      },
      global: {
        stubs: {
          'br-card-content': CardContent,
        },
      },
    })
    expect(wrapper.findComponent(CardContent))
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa a renderização do slot footer.', () => {
    const wrapper = shallowMount(Card, {
      slots: {
        footer: `
          <div slot="footer">
            <br-card-footer>
              <div class="d-flex">
                <div>
                  <button label="Button" />
                </div>
                <div class="ml-auto">
                  <button type="circle">
                    <span slot="button-icon">
                      <i class="fas fa-share-alt" aria-hidden="true"></i>
                    </span>
                  </button>
                  <button type="circle">
                    <span slot="button-icon">
                      <i class="fas fa-heart" aria-hidden="true"></i>
                    </span>
                  </button>
                </div>
              </div>
            </br-card-footer>
          </div>
        `,
      },
      global: {
        stubs: {
          'br-card-footer': CardFooter,
        },
      },
    })

    expect(wrapper.findComponent(CardFooter))
    expect(wrapper.element).toMatchSnapshot()
  })
})
