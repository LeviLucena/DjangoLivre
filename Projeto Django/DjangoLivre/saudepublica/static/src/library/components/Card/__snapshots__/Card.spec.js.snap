// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Testa a renderização do card com Collapse. 1`] = `
<div
  class="br-card"
>
  <div
    class="card-header"
  >
    
    
    <br-list
      class="collapse-wrapper collapsed"
    >
      <br-item
        hover=""
      >
        <div
          class="content"
        >
          <div
            class="flex-fill"
          />
          <icon-base-stub
            class=""
            familyname="fas"
            iconname="chevron-down"
          />
        </div>
      </br-item>
      <div
        class="collapse-content"
        style="display: none;"
      >
        
        <div
          class="br-list mt-3"
        >
          <div
            class="br-item"
          >
            <div
              class="row"
            >
              <div
                class="col-auto"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-heartbeat"
                />
              </div>
              <div
                class="col"
              >
                <p
                  class="m-0"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit.
                </p>
              </div>
            </div>
          </div>
          <span
            class="br-divider"
          />
          <div
            class="br-item"
          >
            <div
              class="row"
            >
              <div
                class="col-auto"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-heartbeat"
                />
              </div>
              <div
                class="col"
              >
                <p
                  class="m-0"
                >
                   Nesciunt, laudantium ea officiis hic tempora velit aperiam nemo accusamus nisi, eligendi ducimus! Incidunt ullam minima ratione amet sequi. 
                </p>
              </div>
            </div>
          </div>
          <span
            class="br-divider"
          />
          <div
            class="br-item"
          >
            <div
              class="row"
            >
              <div
                class="col-auto"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-heartbeat"
                />
              </div>
              <div
                class="col"
              >
                <p
                  class="m-0"
                >
                  Voluptates, iste recusandae.
                </p>
              </div>
            </div>
          </div>
        </div>
        
      </div>
    </br-list>
    
    
  </div>
  <div
    class="card-content"
    tabindex="0"
  >
    
    
  </div>
  <div
    class="card-footer"
  >
    
    
  </div>
</div>
`;

exports[`Card Testa a renderização do slot content. 1`] = `
<div
  class="br-card"
>
  <div
    class="card-header"
  >
    
    
  </div>
  <div
    class="card-content"
    tabindex="0"
  >
    
    <div
      slot="content"
    >
      
      <p>
         Lorem ipsum dolor sit, amet consectetur adipisicing elit. Tempore perferendis nam porro atque ex at, numquam non optio ab eveniet error vel ad exercitationem, earum et fugiat recusandae harum? Assumenda. 
      </p>
      
    </div>
    
  </div>
  <div
    class="card-footer"
  >
    
    
  </div>
</div>
`;

exports[`Card Testa a renderização do slot footer. 1`] = `
<div
  class="br-card"
>
  <div
    class="card-header"
  >
    
    
  </div>
  <div
    class="card-content"
    tabindex="0"
  >
    
    
  </div>
  <div
    class="card-footer"
  >
    
    <div
      slot="footer"
    >
      
      <div
        class="d-flex"
      >
        <div>
          <button
            label="Button"
          />
        </div>
        <div
          class="ml-auto"
        >
          <button
            type="submit"
          >
            <span
              slot="button-icon"
            >
              <i
                aria-hidden="true"
                class="fas fa-share-alt"
              />
            </span>
          </button>
          <button
            type="submit"
          >
            <span
              slot="button-icon"
            >
              <i
                aria-hidden="true"
                class="fas fa-heart"
              />
            </span>
          </button>
        </div>
      </div>
      
    </div>
    
  </div>
</div>
`;

exports[`Card Testa a renderização do slot header. 1`] = `
<div
  class="br-card"
>
  <div
    class="card-header"
  >
    
    <div
      slot="header"
    >
      
      <div
        class="d-flex"
      >
        <span
          class="br-avatar small"
          imgsrc="https://picsum.photos/id/823/400"
        >
          <span
            class="image bg-support-01 text-secondary-01"
          >
            J
          </span>
        </span>
        <div
          class="ml-3"
        >
          <p
            class="h5 text-primary-default mb-0"
          >
            Maria Amorim
          </p>
          <span>
            UX Designer
          </span>
        </div>
        <div
          class="ml-auto"
        >
          <button
            type="submit"
          >
            <span
              slot="button-icon"
            >
              <i
                aria-hidden="true"
                class="fas fa-ellipsis-v"
              />
            </span>
          </button>
        </div>
      </div>
      
    </div>
    
  </div>
  <div
    class="card-content"
    tabindex="0"
  >
    
    
  </div>
  <div
    class="card-footer"
  >
    
    
  </div>
</div>
`;
